### ISSUE TITLE
Allow RTMP streaming to domain name instead of RTMP

### ISSUE THREAD

**Issue by mayeaux** [arut/nginx-rtmp-module#1180]
My domain name is setup using Cloudflare and I'd prefer to use that (aka mydomain.com:1935) as opposed to giving the user the raw IP address of the server

Is this possible? I've read through documentation and a lot online and haven't found anything, and my hack attempts at implementing it aren't working, thanks a lot and thanks for the great module you've made here!

**Comment by tanabi** [comment]
You can't intake RTMP using CloudFlare, but you can broadcast streams using HLS via CloudFlare because HLS is a standard over HTTP protocol.  What I've done in the past is use 2 host names.  For people broadcasting streams, I give them a host name that goes straight to the real machine; for viewers, I direct them to CloudFlare.  Note that this could be against CloudFlare's TOS so if you're doing heavy duty stuff you shouldn't do it through CloudFlare.

**Comment by winshining** [comment]
@mayeaux Yeah, that's possible! I've implemented virtual hosts functionality in RTMP module according to HTTP module: https://github.com/winshining/nginx-http-flv-module.
With it, you can do it as what you do in HTTP module, for example:
    publish: ffmpeg -re -i your_media_file.mp4 -c:v copy -c:v copy -f flv rtmp://**your.domain.xxx**/yourapp/yourstream
    play: rtmp://**your.domain.xxx**/yourapp/yourstream

**Comment by pumpkinpieman** [comment]
@winshining Sorry to ask a silly question here - but where do I place this module?

**Comment by winshining** [comment]
@pumpkinpieman You mean in the Makefile?
